src/Network/Ricochet/Protocol/Lowest.hs

```diff
@@ -2,31 +2,36 @@
 module Network.Ricochet.Protocol.Lowest
   ( parsePacket
   , dumpPacket
-  , parseWord16
   , splitInPackets
   ) where
 
-import           Data.Bifunctor          (first)
-import           Data.Bits               (shiftL)
-import           Data.ByteString         (ByteString ())
-import qualified Data.ByteString         as B
-import           Data.ByteString.Builder (byteString, toLazyByteString,
-                                          word16BE)
-import           Data.ByteString.Lazy    (toStrict)
-import           Data.Monoid             ((<>))
-import           Data.Word               (Word16, Word8)
-import           Network.Ricochet.Types  (Packet (..))
-
--- | Parses the low-level representation of a Packet.
-parsePacket :: ByteString -> Maybe (Packet, ByteString)
-parsePacket bs = case parseWord16 bs of
-    Just (size, bs') -> case parseWord16 bs' of
-      Just (channelId, bs'') -> case (toInteger . B.length $ bs'') `compare` toInteger size of
-        EQ -> Just . (,B.empty) $ MkPacket size channelId bs''
-        GT -> Just $ (MkPacket size channelId $ B.take (fromIntegral size) bs'', B.drop (fromIntegral size) bs'')
-        LT -> Nothing
-      Nothing -> Nothing
-    Nothing -> Nothing
+import           Prelude                    hiding (take)
+
+import           Network.Ricochet.Types     (Packet (..), makePacket)
+import           Network.Ricochet.Util
+
+import           Data.Attoparsec.ByteString
+import           Data.Bifunctor             (first)
+import           Data.ByteString            (ByteString ())
+import qualified Data.ByteString            as B
+import           Data.ByteString.Builder    (byteString, toLazyByteString,
+                                             word16BE)
+import           Data.ByteString.Lazy       (toStrict)
+import           Data.Monoid                ((<>))
+import           Data.Word                  (Word16, Word8)
+
+
+-- | Actually parses a packet.
+parsePacket :: ByteString -> Maybe (Maybe (Packet, ByteString))
```

Comment:
I’m not sure what the `Maybe (Maybe (…))` return value does.  This should be changed or mentioned in the comment.